### PR TITLE
Merge OrientedPathFollow into PathFollow

### LINKED ISSUE #23070: OrientedPathFollow should be renamed PathFollowOriented
**Godot version:** 3.1 Alpha and master since May 25

**Issue description:** See https://github.com/godotengine/godot/pull/18752

The new OrientedPathFollow should be renamed PathFollowOriented, so that all of the Path nodes start with the word Path. This node isn't present in 3.0 so renaming doesn't break compatibility.

However, it might even be better for PathFollow to have an "Oriented" boolean or enum instead.

Suggested Milestone: 3.1. A decision should be made on this issue before 3.1 Beta.

### PR DESCRIPTION
**Pull request by davi2205**
As suggested by @aaronfranke in #23070.

*Bugsquad edit:* Fixes #23070.

### CODE REVIEW

**Comment by akien-mga** [comment]
I think it would be better to drop `OrientedPathFollow` altogether and make the "oriented" mode a boolean flag in `PathFollow`, as suggested in #23070.

**Comment by reduz** [comment]
Agreed, it should just be a bool in PathFollow

**Comment by danilo2205** [comment]
Ok, I'll be push-ing the changes asap.

**Comment by danilo2205** [comment]
I was thinking about creating an extra item in `PathFollow`'s `RotationMode` enum since `PathFollowOriented` does not use `RotationMode` at all, it could replace the suggested extra bool property.

image

**Comment by akien-mga** [comment]
@danilo2205 reduz confirms that it's a good suggestion, so go for it :)

**Comment by danilo2205** [comment]
Changes made:

- Removed `OrientedPathFollow` completely.
- Added `ROTATION_ORIENTED` to the rotation_mode property of `PathFollow`.
- `ROTATION_ORIENTED` behaves exactly the same as `OrientedPathFollow`.
- Removed documentation and icon of `OrientedPathFollow`.
- Added documentation for the new `ROTATION_ORIENTED`.

**Review by aaronfranke — APPROVED** [review]

File: doc/classes/Curve3D.xml [hunk 1]
```diff
@@ -248,7 +248,7 @@
 			The distance in meters between two adjacent cached points. Changing it forces the cache to be recomputed the next time the [method get_baked_points] or [method get_baked_length] function is called. The smaller the distance, the more points in the cache and the more memory it will consume, so use with care.
 		</member>
 		<member name="up_vector_enabled" type="bool" setter="set_up_vector_enabled" getter="is_up_vector_enabled">
-			If [code]true[/code], the curve will bake up vectors used for orientation. See [OrientedPathFollow]. Changing it forces the cache to be recomputed.
+			If [code]true[/code], the curve will bake up vectors used for orientation. This is used when a [PathFollow]'s [code]rotation_mode[/code] is set to [code]ROTATION_ORIENTED[/code], see [PathFollow] for details. Changing it forces the cache to be recomputed.
```
Comment (akien-mga) [inline doc/classes/Curve3D.xml]: You can use `[member PathFollow.rotation_mode]` to create a direct hyperlink to that property.

File: doc/classes/PathFollow.xml [hunk 2]
```diff
@@ -51,5 +51,8 @@
 		<constant name="ROTATION_XYZ" value="3" enum="RotationMode">
 			Allows the PathFollow to rotate in any axis.
 		</constant>
+		<constant name="ROTATION_ORIENTED" value="4" enum="RotationMode">
```
Comment (akien-mga) [inline doc/classes/PathFollow.xml]: Same here, you can use `[member Curve3D.up_vector_enabled]`.

File: doc/classes/Curve3D.xml (hunk 1 above)
Comment (danilo2205) [reply doc/classes/Curve3D.xml]: ok

File: doc/classes/PathFollow.xml (hunk 2 above)
Comment (danilo2205) [reply doc/classes/PathFollow.xml]: ok

**Comment by danilo2205** [comment]
Just fixed the documentation.

**Review by akien-mga — APPROVED** [review]

**Pull request merged**